__tests__/core/jsonpatch.test.ts

```diff
@@ -16,7 +16,7 @@ import {
   joinJsonPath
 } from "../../src"
 
-function testPatches<C, S, T>(
+function testPatches<C, S, T extends object>(
```

Comment:
This should be correct. Without it, `getSnapshot` is complaining:

```
Argument of type 'STNValue<T, IType<C, S, T>>' is not assignable to parameter of type 'IStateTreeNode<IType<any, unknown, any>>'.
  Type 'T | (T & IStateTreeNode<IType<C, S, T>>)' is not assignable to type 'IStateTreeNode<IType<any, unknown, any>>'.
    Type 'T' is not assignable to type 'IStateTreeNode<IType<any, unknown, any>>'.ts(2345)
jsonpatch.test.ts(19, 28): This type parameter might need an `extends IStateTreeNode<IType<any, unknown, any>>` constraint.
```

The problem is that `IType["create"]` returns an `STNValue`, and without any constraints you can see that it will be `(T & IStateTreeNode<IT>) | T`, and `getSnapshot` needs an `IStateTreeNode<IT>`.

Reply:
Thanks for the note! This makes sense. I think we were hitting that over in https://github.com/mobxjs/mobx-state-tree/pull/2103